crates/holochain/src/core/sys_validate.rs

```diff
@@ -421,18 +423,26 @@ pub fn check_tag_size(tag: &LinkTag) -> SysValidationResult<()> {
 /// Check a Update's entry type is the same for
 /// original and new entry.
 pub fn check_update_reference(
-    eu: &Update,
+    update: &Update,
     original_entry_action: &NewEntryActionRef<'_>,
 ) -> SysValidationResult<()> {
-    if eu.entry_type == *original_entry_action.entry_type() {
-        Ok(())
-    } else {
-        Err(ValidationOutcome::UpdateTypeMismatch(
+    if update.entry_type != *original_entry_action.entry_type() {
+        return Err(ValidationOutcome::UpdateTypeMismatch(
             original_entry_action.entry_type().clone(),
-            eu.entry_type.clone(),
+            update.entry_type.clone(),
         )
-        .into())
+        .into());
+    }
+
+    if update.original_entry_address != *original_entry_action.entry_hash() {
```

Comment:
I was surprised this wasn't being checked. We're implicitly checking the `original_action_hash` because we're looking it up. So you have to provide a value that points at some previous if you want this op to pass validation. However, I'd expect the entry to be valid if I got an op that had passed validation. Yes I can always look up the previous action and check its entry hash, but there shouldn't be a difference I believe?